model/task_lifecycle.go

```diff
@@ -77,16 +77,6 @@ func SetActiveState(caller string, active bool, tasks ...task.Task) error {
 			// If the task was not activated by step back, and either the caller is not evergreen
 			// or the task was originally activated by evergreen, deactivate the task
 		} else if !evergreen.IsSystemActivator(caller) || evergreen.IsSystemActivator(t.ActivatedBy) {
-			// deactivate later tasks in the group as well, since they won't succeed without this one
```

Comment:
I read through the reasoning in the ticket, but I think the hypothesis about `DeactivateDependencies` deactivating single-host task groups is incorrect. I believe (@ablack12 double-check me that this is still true?) that the dependencies are absent when the tasks are first created, and the dependency chain between single-host task group tasks is only retroactively created _after_ the task runs and fails (using [BlockTaskGroupTasks](https://github.com/evergreen-ci/evergreen/blob/033f687de1f50c9c50f22a4e4090ee0e9661a1be/model/task_queue.go#L240-L293) to make the dependencies and mark them unattainable). Therefore, I would imagine that just blocking dependencies isn't guaranteed to block a single-host task group if it hasn't run yet. If we can't guarantee that single-host task groups have the dependencies set up when deactivating/disabling, I think we have to keep this logic.

Reply:
cc @hadjri you may want to decide if this is out of scope of the project or not (I didn't read closely into it but can discuss if it's not clear)

Reply:
I suppose it's reasonable to include this in the project since it has spawned off of this PR and is still relevant to the original ticket.